docs/pages/guides/onchain-indices.mdx

```diff
@@ -0,0 +1,412 @@
+import { CollapseCode } from "../../components/CollapseCode";
+
+# Onchain Indices
+
+This guide walks through why you might need onchain indices, and how you can create and manage them.
+
+## What are indices?
+
+In MUD, we can answer simple queries onchain like _“what is Alice’s position?”_ with `Position.get(alice)`.
+
+However, we often have use for more complex transformations of onchain data that are infeasible to compute onchain. For example, _“which players are at this position?”_, or _“how many gold mines does Bob own?_.
+
+For this reason, we create onchain “indices” that maintain the result of these transformations.
+
+## Indices in MUD
+
+In these examples, we will make indices for a simple game where each account has an (x, y) position.
+
+```ts filename="mud.config.ts" copy showLineNumbers
+import { mudConfig } from "@latticexyz/world/register";
+
+export default mudConfig({
+  tables: {
+    Position: {
+      keySchema: {
+        player: "address",
+      },
+      valueSchema: {
+        x: "int32",
+        y: "int32",
+      },
+    },
+  },
+});
+```
+
+In this game, players update their position by calling `MoveSystem.move` with their desired coordinates.
+
+```solidity filename="MoveSystem.sol" copy showLineNumbers
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { System } from "@latticexyz/world/src/System.sol";
+import { Position } from "../codegen/index.sol";
+
+contract MoveSystem is System {
+  function move(int32 x, int32 y) public {
+    address player = _msgSender();
+
+    Position.set(player, x, y);
+  }
+}
+```
+
+To make the game more interesting, we would like to prevent players from moving to positions that are already occupied. However, it's not clear how to tell which players are at a given position onchain.
+
+   <CollapseCode>
+
+```solidity filename="MoveSystem.sol" copy showLineNumbers {11-14}
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { System } from "@latticexyz/world/src/System.sol";
+import { Position } from "../codegen/index.sol";
+
+contract MoveSystem is System {
+  function move(int32 x, int32 y) public {
+    address player = _msgSender();
+
+    bool positionIsEmpty = ?
+    require(positionIsEmpty, "another player is at this position");
+
+    Position.set(player, x, y);
+  }
+}
+```
+
+   </CollapseCode>
+
+In other words, there is no reverse mapping - we can read the position of a given player, but not which players are at each position. It's time to add an index!
+
+There are two main ways to implement this - you can **create and manage your own indices**, or **use a module**.
+
+### Creating your own indices
+
+#### `PlayersAtPosition`
```

Comment:
I think it would be best to remove this example. It's complicated and as you wrote in `PositionIsOccupied` example, needlessly so. My comments in this section are in case you decide to keep it.

Reply:
Agreed that having both an `address` and `bool` mapping is pointless!